lib/handler/cache-handler.js

```diff
@@ -75,14 +75,14 @@ class CacheHandler extends DecoratorHandler {
     )
 
     if (
-      UNSAFE_METHODS.includes(this.#req.method) &&
+      !this.#methods.includes(this.#requestOptions.method) &&
```

Comment:
WHy did we check for unsafe methods? this whole if condition seems actually kind of dubious.

Reply:
I would have expected 

```js
    if (
      !this.#methods.includes(this.#requestOptions.method) ||
      (
        statusCode < 200 &&
        statusCode >= 400
      )
    ) {
```

Reply:
@flakey5 

The invalidation logic is kind of strange. Also what is the point of passing methods, when they are ignored anyway? So I guess actually methods would have been an array of safe methods?

Reply:
> A cache MUST invalidate the target URI ([Section 7.1](https://www.rfc-editor.org/rfc/rfc9110#section-7.1) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9111.html#HTTP)]) when it receives a non-error status code in response to an unsafe request method (including methods whose safety is unknown).

So I guess my proposed change should be right, as the passed methods will be considered safe methods. Especially in regard of the last sentence in the spec, that methods whose safety is unknown is also covered.